tfhe/src/safe_serialization.rs

```diff
@@ -0,0 +1,607 @@
+    /// the current *TFHE-rs* version.
+    pub fn new(serialized_size_limit: u64) -> Self {
+        Self {
+            serialized_size_limit,
+            validate_header: true,
+        }
+    }
+
+    /// Creates a new config without any size limit for the deserialized objects.
+    pub fn new_with_unlimited_size() -> Self {
+        Self {
+            serialized_size_limit: 0,
+            validate_header: true,
+        }
+    }
+
+    /// Disables the size limit for the serialized objects.
+    pub fn disable_size_limit(self) -> Self {
+        Self {
+            serialized_size_limit: 0,
+            ..self
+        }
+    }
+
+    /// Disables the header validation on the object. This header validations
+    /// checks that the serialized object is the one that is supposed to be loaded
+    /// and is compatible with this version of *TFHE-rs*.
+    pub fn disable_header_validation(self) -> Self {
+        Self {
+            validate_header: false,
+            ..self
+        }
+    }
+
+    /// Disables the conformance check on an existing config.
+    pub fn disable_conformance(self) -> NonConformantDeserializationConfig {
+        NonConformantDeserializationConfig {
+            serialized_size_limit: self.serialized_size_limit,
+            validate_header: self.validate_header,
+        }
+    }
+
+    /// Deserializes an object serialized by [`SerializationConfig::serialize_into`] from a
+    /// [reader](std::io::Read). Performs various sanity checks based on the deserialization config.
+    pub fn deserialize_from<T: DeserializeOwned + Unversionize + Named + ParameterSetConformant>(
+        self,
+        reader: impl std::io::Read,
+        parameter_set: &T::ParameterSet,
+    ) -> Result<T, String> {
+        let deser: T = self.disable_conformance().deserialize_from(reader)?;
+        if !deser.is_conformant(parameter_set) {
+            return Err(format!(
+                "Deserialized object of type {} not conformant with given parameter set",
+                T::NAME
+            ));
+        }
+
+        Ok(deser)
+    }
+}
+
+pub fn safe_serialize<T: Serialize + Versionize + Named>(
+    object: &T,
+    writer: impl std::io::Write,
+    serialized_size_limit: u64,
+) -> bincode::Result<()> {
+    SerializationConfig::new(serialized_size_limit).serialize_into(object, writer)
+}
+
+pub fn safe_deserialize<T: DeserializeOwned + Unversionize + Named>(
+    reader: impl std::io::Read,
+    serialized_size_limit: u64,
+) -> Result<T, String> {
+    DeserializationConfig::new(serialized_size_limit)
+        .disable_conformance()
+        .deserialize_from(reader)
+}
+
+pub fn safe_deserialize_conformant<
+    T: DeserializeOwned + Unversionize + Named + ParameterSetConformant,
+>(
+    reader: impl std::io::Read,
+    serialized_size_limit: u64,
+    parameter_set: &T::ParameterSet,
+) -> Result<T, String> {
+    DeserializationConfig::new(serialized_size_limit).deserialize_from(reader, parameter_set)
+}
+
```

Comment:
Do we still want these functions?
They have been removed from many places (if not all)

Reply:
It was asked by BC users to keep those in this release to simplify their migration, but it can be removed in the next release. I can add a deprecation attribute if we want to. 

However we might also decide that they serve as a slightly easier shortcut for users that don't want to do any configuration.